Tests/Marketplace/pack_readme_handler.py

```diff
@@ -0,0 +1,223 @@
+
+        # no external image urls were found in the readme file
+        if not readme_images_storage_paths:
+            logging.info(f'no image links were found in {pack_name} readme file')
+            continue
+
+        pack_readme_images_list = [image_info.get('image_name') for image_info in readme_images_storage_paths]
+        readme_images[pack_name] = pack_readme_images_list
+        logging.info(f'Added {pack_readme_images_list=} from pack {pack_name=} to the artifact dict')
+        readme_urls_data_list = readme_urls_data_list + readme_images_storage_paths
+
+    return readme_images, readme_urls_data_list
+
+
+def collect_images_from_readme_and_replace_with_storage_path(pack_readme_path: str, gcs_pack_path, pack_name: str,
+                                                             marketplace: str = 'xsoar', index_v2: bool = False) -> list:
+    """
+    Replaces inplace all images links in the pack README.md with their new gcs location
+
+    Args:
+        pack_readme_path (str): A path to the pack README file.
+        gcs_pack_path (str): A path to the pack in gcs
+        marketplace (str): The marketplace this pack is going to be uploaded to.
+
+    Returns:
+        A list of dicts of all the image urls found in the README.md file with all related data
+        (original_url, new_gcs_path, image_name)
+    """
+    if index_v2:
+        google_api_readme_images_url = f'api/marketplace/file?name=content/packs/{pack_name}'
+    else:
+        marketplace_bucket = (
+            "marketplace-dist"
+            if marketplace == 'xsoar'
+            else "marketplace-v2-dist"
+        )
+        google_api_readme_images_url = f'https://storage.googleapis.com/{marketplace_bucket}/content/packs/{pack_name}'
+
+    url_regex = r"^!\[(.*)\]\((?P<url>.*)\)"
+    urls_list = []
+
+    for line in fileinput.input(pack_readme_path, inplace=True):
+        if res := re.search(url_regex, line):
+            url = res['url']
+
+            parse_url = urllib.parse.urlparse(url)
+            path = parse_url.path
+            url_path = Path(path)
+            image_name = url_path.name
+
+            new_replace_url = os.path.join(google_api_readme_images_url,
+                                           BucketUploadFlow.README_IMAGES, image_name)
+            logging.info(f'replacing with url {new_replace_url=}')
+            line = line.replace(url, str(new_replace_url))
+
+            image_gcp_path = Path(gcs_pack_path, BucketUploadFlow.README_IMAGES, image_name)
+            urls_list.append({
+                'original_read_me_url': url,
+                'new_gcs_image_path': image_gcp_path,
+                'image_name': image_name
+            })
+
+        print(line, end='')
+
+    return urls_list
+
+
+def download_readme_images_from_url_data_list(readme_urls_data_list: list, storage_bucket):
+    """
+    Iterates over the readme_url_data_list and calls the download_readme_image_from_url_and_upload_to_gcs
```

Comment:
Add description for the function arguments

Reply:
done